server/game/stateWatchers/CardsLeftPlayThisPhaseWatcher.ts

```diff
@@ -34,6 +37,22 @@ export class CardsLeftPlayThisPhaseWatcher extends StateWatcher<CardLeftPlayEntr
             .map((entry) => entry.card);
     }
 
+    public someCardsLeftPlayControlledByPlayer({ controller, filter }: {
+        controller: Player;
+        filter?: (event: CardLeftPlayEntry) => boolean;
+    }) {
+        const playerFilter = (entry: CardLeftPlayEntry) => entry.controlledBy === controller;
+
+        if (filter != null) {
```

Comment:
We should just call `getCardsLeftPlayControlledByPlayer()` and checking the result instead of repeating the code, same for similar methods in other watchers